core/request.py

```diff
@@ -298,16 +299,24 @@ def processRequest(self) -> None:
                     logging.exception(newE)
                     res = None
             if not res:
-                with (conf["viur.instance.core_base_path"]
-                      .joinpath(conf["viur.errorTemplate"])
-                      .open() as tpl_file):
-                    tpl = Template(tpl_file.read())
-                res = tpl.safe_substitute({
-                    "error_code": e.status,
-                    "error_name": translate(e.name),
-                    "error_descr": e.descr,
-                })
-                extendCsp({"style-src": ['sha256-Lwf7c88gJwuw6L6p6ILPSs/+Ui7zCk8VaIvp8wLhQ4A=']})
+                if (len(self.pathlist) > 0 and any(x in self.pathlist[0] for x in ["vi", "json"])) or \
+                        utils.currentRequest.get().response.headers["Content-Type"] == "application/json":
+                    utils.currentRequest.get().response.headers["Content-Type"] = "application/json"
```

Comment:
This line can be removed, as the content-type is already "application/json"
```suggestion
```

Reply:
But not necessarily in the case `len(self.pathlist) > 0 and any(x in self.pathlist[0] for x in ["vi", "json"]))` Or?

Reply:
You're right!